packages/esbuild/src/executors/esbuild/lib/build-esbuild-options.ts

```diff
@@ -123,6 +127,50 @@ export function buildEsbuildOptions(
   return esbuildOptions;
 }
 
+/**
+ * When using TS project references we need to map the paths to the referenced projects.
+ * This is necessary because esbuild does not support project references out of the box.
+ * @param context ExecutorContext
+ */
+export function createPathsFromTsConfigReferences(
+  context: ExecutorContext
+): Record<string, string[]> {
+  const {
+    findAllProjectNodeDependencies,
+  } = require('nx/src/utils/project-graph-utils');
+
+  const deps = findAllProjectNodeDependencies(context.projectName);
```

Comment:
nit: we can provide the project graph we already have available to avoid the operation of retrieving it from the cache unnecessarily.
```suggestion
  const deps = findAllProjectNodeDependencies(
    context.projectName,
    context.projectGraph
  );
```